tutorials/shaders/shader_reference/shader_functions.rst

```diff
@@ -0,0 +1,5273 @@
+|              |                                                                                                     | Not available on ``gl_compatibility`` profile.                      |
++--------------+-----------------------------------------------------------------------------------------------------+---------------------------------------------------------------------+
+| |vec_type|   | :ref:`fwidthFine<shader_func_fwidthFine>` ( |vec_type| p)                                           | Sum of absolute derivative in ``x`` and ``y``.                      |
+|              |                                                                                                     | Not available on ``gl_compatibility`` profile.                      |
++--------------+-----------------------------------------------------------------------------------------------------+---------------------------------------------------------------------+
+
+.. rst-class:: classref-section-separator
+
+----
+
+
+
+
+.. _shader_func_textureSize:
+
+.. rst-class:: classref-method
+
+ivec2 **textureSize** ( |gsampler2D| s, int lod )
+
+    Retrieve the dimensions of a level of a texture.
+
+    Returns the dimensions of level lod (if present) of the texture bound to sampler.
+
+    The components in the return value are filled in, in order, with the width, height and depth
+    of the texture. For the array forms, the last component of the return value is
+    the number of layers in the texture array.
+
+    :param s:
+        the sampler to which the texture whose dimensions to retrieve is bound.
+
+    :param lod:
+        the level of the texture for which to retrieve the dimensions.
+
+    :return:
+        the dimensions of level lod (if present) of the texture bound to sampler.
+
+    https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/textureSize.xhtml
+
+.. rst-class:: classref-item-separator
+
+----
+
+
+.. rst-class:: classref-method
+
+ivec2 **textureSize** ( samplerCube s, int lod )
+
+    Retrieve the dimensions of a level of a texture.
+
+    Returns the dimensions of level lod (if present) of the texture bound to sampler.
+
+    The components in the return value are filled in, in order, with the width, height and depth
+    of the texture. For the array forms, the last component of the return value is
+    the number of layers in the texture array.
+
+    :param s:
+        the sampler to which the texture whose dimensions to retrieve is bound.
+
+    :param lod:
+        the level of the texture for which to retrieve the dimensions.
+
+    :return:
+        the dimensions of level lod (if present) of the texture bound to sampler.
+
+    https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/textureSize.xhtml
+
+.. rst-class:: classref-item-separator
+
+----
+
+
+.. rst-class:: classref-method
+
+ivec2 **textureSize** ( samplerCubeArray s, int lod )
```

Comment:
The following text is just a duplicated information, I think you should remove it, and instead add the overloads to header of first declaration eg:

```
ivec2 **textureSize** ( |gsampler2D| s, int lod )
ivec2 **textureSize** ( samplerCube s, int lod )
ivec3 **textureSize** ( |gsampler2DArray| s, int lod )
ivec3 **textureSize** ( |gsampler3D| s, int lod )
```

And do the same for other functions...

Reply:
There were a number of these kinds of functions, in a few of them the text was exactly the same, but a lot of them had minor differences. For instance, the `atan` function has significant behavioral differences between its two overloads. 

I decided that the documentation would be slightly clearer if these were addressed individually and that consistency of layout/format (in this case meaning each overload gets its own documentation) was more important than concerns about repetition.

Admittedly, most of them are not-just the difference between int and uint or similarly small, but even some of these behave differently, such as `packUnorm2x16` which has a different snippet of equivalent code for each overload.

Reply:
@Chaosus thoughts?

Reply:
I think we can separate them in cases like atan where you need two different descriptions. But in cases like `textureSize`, it makes sense to group them together. You are right not to worry about repetition, but some cases (like all the `texture*` functions) the sheer amount of repetition makes the document harder to navigate

Reply:
Understood, I will add this to my todo list.